### PR TITLE
Settings updates

### LINKED ISSUE #229: Extract blade component
Refactor certain blade.php components into separate sub pieces so that we can simply reuse them instead of having to update in multiple places. 
### What needs to be done
 
`resources/views/app.blade.php` and `resources/views/layouts/guest.blade.php ` have the exact same code and can be extracted.

### LINKED ISSUE #447: Settings Title affect header of navbar
Have the settings apply to the header
### What needs to be done
 
Just add the check on each page like i do for header image.

### LINKED ISSUE #448: Change hashing method for file saved for app logo
Theoretically 2 users submitting a photo at the same time can cause problems. We will resolve this by hashing the filename instead of just using the time and date. 
### What needs to be done
 
apply a hash using time and file name.

### PR DESCRIPTION
**Pull request by EvanDime**
##### **_IMPORTANT: Always create an issue before a PR and provide enough information so that others can review_**

Adding fix for app_logo file save and use app_title for page header instead of just the tab title. 
##### _What existing problem does the pull request solve?_

Makes app_logo more secure.
Adds more relevance to the app title feature. 
##### _Demonstrate the code is solid. </br> Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI._

Update Settings test to check for the correct hash. 
Adding Frontend test to check if the title displays. 
##### _List all issues this PR resolves_
- closes #448 
- closes #447 
- closes #229 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Integration Tests
- [x] System Tests
- [x] Documentation

### CODE REVIEW

File: resources/views/app_head.blade.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<meta charset="utf-8">
```
Comment (alexstojda) [inline resources/views/app_head.blade.php:1]: Good idea moving this to a new component

File: app/Http/Middleware/HandleInertiaRequests.php [hunk 2]
```diff
@@ -38,12 +38,13 @@ public function version(Request $request)
      */
     public function share(Request $request)
     {
-        Config::set('app.name', Settings::where('slug', 'app_name')->first()->data['name'] ?? 'Laravel');
-
-        $logo = Settings::where('slug', 'app_logo')->first()->data['path'] ?? null;
+      $name = Settings::where('slug', 'app_name')->first()->data['name'] ?? 'CSU Booking Platform';
```
Comment (alexstojda) [inline app/Http/Middleware/HandleInertiaRequests.php:41]: Default values are always a good idea

File: app/Http/Controllers/SettingsController.php [hunk 3]
```diff
@@ -46,22 +47,22 @@ public function storeAppName(Request $request)
...
-            'logos',
-                $request->file('app_logo'),
-                now()->format('d-m-y-H-i-s').'.'.$request->file('app_logo')->extension());
-        Settings::updateOrCreate(
-            ['slug' => $request->label],
-            ['data' => ['path' => $path]]
-        );
-        return back();
-    }
+  public function storeAppLogo(Request $request)
```
Comment (alexstojda) [inline app/Http/Controllers/SettingsController.php:50]: 2-space indents are indeed much easier to read for narrow-screens (and in github's UI)

**Review by alexstojda — APPROVED** [review]
Looks good evan!

File: app/Http/Controllers/SettingsController.php (hunk 3 above)
Comment (WindOfRussia) [reply app/Http/Controllers/SettingsController.php:50]: true but also PSR2

File: app/Http/Controllers/SettingsController.php [hunk 4]
```diff
@@ -46,22 +47,22 @@ public function storeAppName(Request $request)
...
-    public function storeAppLogo(Request $request)
-    {
-        $request->validate([
-            'app_logo' => 'required|image|mimes:jpg,png,jpeg,gif,svg|max:2048',
-            'label' => 'required'
-        ]);
-        $path = 'storage/' . Storage::disk('public')->putFileAs(
-            'logos',
-                $request->file('app_logo'),
-                now()->format('d-m-y-H-i-s').'.'.$request->file('app_logo')->extension());
```
Comment (WindOfRussia) [inline app/Http/Controllers/SettingsController.php:58]: hey is there an already existing format you could use? carbon has a lot of them prebuit -> toDateString, toDatetimeString, toISO...String.

File: tests/Feature/SettingsControllerTest.php [hunk 5]
```diff
@@ -88,11 +88,11 @@ public function testFormUpdateCreateAppLogo()
             'label' => 'app_logo',
             'app_logo' => $file,
         ]);
-        Storage::disk('public')->assertExists('logos/' . now()->format('d-m-y-H-i-s').'.'.$file->extension());
+        Storage::disk('public')->assertExists('logos/' . $file->hashName());
```
Comment (WindOfRussia) [inline tests/Feature/SettingsControllerTest.php:91]: good call simplifying these concat statements

File: tests/Browser/SettingsPageTest.php [hunk 6]
```diff
@@ -31,13 +31,15 @@ public function testAdminCanChangeSiteName()
   {
     $this->browse(function (Browser $browser) {
       $browser->loginAs(User::first())->visit('/admin/settings')
-        ->assertSourceHas('<title>Laravel</title>')
+        ->assertSourceHas('<title>CSU Booking Platform</title>')
```
Comment (WindOfRussia) [inline tests/Browser/SettingsPageTest.php:34]: maybe you should verify this against the config name or the value from settings db table?

**Review by WindOfRussia — APPROVED** [review]
Added a couple suggestions for improvement but I'll approve. 
@EvanDime up to you if you want to make those changes or not, the PR fulfils its purpose either way.

File: tests/Browser/SettingsPageTest.php (hunk 6 above)
Comment (EvanDime) [reply tests/Browser/SettingsPageTest.php:34]: We already have a test for the backend. This isolates the test to the front end.

**Pull request merged**